MementoMori/mementomori.client/src/CardEditor.tsx

```diff
@@ -0,0 +1,221 @@
+        }
+
+        const dynamicTextValidationError = validateDynamicText(dynamicInputValue);
+        if (dynamicTextValidationError) {
+            setDynamicTextError(dynamicTextValidationError);
+            return;
+        } else {
+            setDynamicTextError('');
+        }
+        if (cardIdField.length === 0)
+        {
+            setDeckIdError('Deck Id cannot be empty');
+        }
+        else
+            setDeckIdError('');
+
+        try {
+            // Create an object to send to the backend
+            const cardData = {
+                deckId: cardIdField,
+                tags: staticInputValue,
+                text: dynamicInputValue,
+            };
+
+            // Send POST request to the server
+            const response = await axios.post('/CardData/postCard', cardData);
+
+            if (response.status === 200) {
+                console.log("Card data successfully saved");
+                setRequestError('Woooho');
+
+                setPostedTags(response.data.tags);
+                setPostedText(response.data.text);
+
+                // Optionally, you can clear the form here or give feedback to the user
+            } else {
+                setRequestError('Error: Failed to save card data');
+            }
+        } catch (error) {
+            if (error.response && error.response.status === 409) {
+                // Handle the case where the file already exists
+                setRequestError('Error: The file "CardInfo.txt" already exists.');
+            } else {
+                // Handle other errors
+                setRequestError('Error: An unexpected error occurred. Error message: ' + error.message);
+            }
+        }
+    };
+
+    return (
+        <Box
+            sx={{
+                minWidth: '50vw',
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+                margin: 'auto',
+            }}
+        >
+            <Card variant="outlined" sx={{ width: '100%', height: 'auto', minHeight: '30vh' }}>
+                <CardContent>
+                    {/* Existing input fields and logic */}
+                    <Typography gutterBottom sx={{ color: 'text.secondary', fontSize: 14 }}>
+                        Enter the Id of an exsisting Deck
+                    </Typography>
+                    <TextField
+                        fullWidth
+                        variant="outlined"
+                        onChange={handleCardIdField}
+                        label="Deck Id"
+                        sx={{
+                            marginBottom: '16px',
+                        }}
+                        value={cardIdField}  // Reset field on form submission
+                        error={Boolean(deckIdError)}
+                        helperText={deckIdError}
+                    />
+                    <Typography gutterBottom sx={{ color: 'text.secondary', fontSize: 14 }}>
```

Comment:
just use `TagSelector.tsx` component here, will be better user experience and will not require to verify input later.